llvm/lib/Transforms/Instrumentation/ThreadSanitizer.cpp

```diff
@@ -752,11 +752,12 @@ bool ThreadSanitizer::instrumentAtomic(Instruction *I, const DataLayout &DL) {
     const unsigned ByteSize = 1U << Idx;
     const unsigned BitSize = ByteSize * 8;
     Type *Ty = Type::getIntNTy(IRB.getContext(), BitSize);
-    Value *Args[] = {Addr,
-                     IRB.CreateIntCast(RMWI->getValOperand(), Ty, false),
+    Value *Val = RMWI->getValOperand();
+    Value *Args[] = {Addr, IRB.CreateBitOrPointerCast(Val, Ty),
                      createOrdering(&IRB, RMWI->getOrdering())};
-    CallInst *C = CallInst::Create(F, Args);
-    ReplaceInstWithInst(I, C);
+    Value *C = IRB.CreateCall(F, Args);
+    I->replaceAllUsesWith(IRB.CreateBitOrPointerCast(C, Val->getType()));
```

Comment:
Looks like `replaceAllUsesWith/eraseFromParent` matches `ReplaceInstWithInst`
However, could we keep code consistent?
Replacing all places in a separate patch LGTM.

Reply:
The reason I changed this is that we now go through IRBuilder, which will insert the instructions itself, so we can no longer use ReplaceInstWithInst. This is also why the cmpxchg code does the same.

Reply:
Converted the remaining uses in https://github.com/llvm/llvm-project/commit/8a237ab7d9022d24441544ba25be480f0c944f5a.

Reply:
Thanks!